src/util/algoliaUtils.js

```diff
@@ -76,9 +79,25 @@ export function rebuildSortByIndexFromRoute(routeStateSortBy) {
 	return setSortByEnv(selectedRouteSort);
 }
 
+// Build object of external query params. ie utm params
+export function extractExternalParams(search) {
+	const searchParams = search.replace('?', '').split('&');
+	// extract utms
+	const utmParamsFromSearch = searchParams.filter(param => param.indexOf('utm_') !== -1);
+	const utmArray = utmParamsFromSearch.map(utmString => {
+		const utm = utmString.split('=');
+		return utm;
+	});
+	return Object.fromEntries(utmArray);
+}
+
 // Build Algolia specific map of current search state to url query parameters
 export function stateToRoute(uiState) {
-	// console.log(`uiState: ${JSON.stringify(uiState)}`);
+	// extract and assign utm params if present
+	if (window.location.search.indexOf('utm') !== -1 && !utmParams.length) {
```

Comment:
I verified that both stateToRoute and routeToState "know" about the windon.location.search value during page load and throughout their first processing cycle. routeToState runs first and will establish the cached set of utm params if they are present, then stateToRoute will use them.